microprofile/WFLY-13048_upgrade_microprofile-health_2.2.adoc

```diff
@@ -0,0 +1,59 @@
+= Upgrade MicroProfile Health to 2.2
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          microprofile,health,observability
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+
+MicroProfile 3.3 provides a minor upgrade for https://github.com/eclipse/microprofile-health/releases/tag/2.2[MicroProfile Health 2.2].
+
+This component upgrade also covers the upgrade of smallrye-health that implements MicroProfile Health 2.2.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13048[WFLY-13048]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1426[EAP7-1426]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+[X] QE
+
+== Requirements
+
+* Upgrade MicroProfile Health to 2.2
+* Upgrade smallrye-health to the version that implements MicroProfile Health 2.2
+* Run any MicroProfile Health test in WildFly integration test suite
+* Run the Eclipse MicroProfile Health TCK without failures
+
+=== Non-Requirements
+
+* Support of the `mp.health.disable-default-procedures` config property is handled separately by to https://issues.jboss.org/browse/WFLY-12952[WFLY-12952] that will add such a default server procedure for the readiness check.
+
+== Implementation Plan
+
+== Test Plan
+
+* Run the WildFly integration basic test suite and checks there are no failures related to MicroProfle Health tests.
+* Run the `microprofile-tck/health` module from WildFly integration test suite and checks that there are no failures.
+
```

Comment:
Are new features - e.g.: 

* enhancement of REST Client integration, see https://github.com/eclipse/microprofile-health/pull/218/files

* annotation literals, see https://github.com/eclipse/microprofile-health/pull/225

covered by basic test cases in Wildfly TS/TCK? I'd add a mention to the general topic of coverage for the new features. WDYT?

Reply:
* annotation literals is mostly an implementation detail that does not accept consumer of the API. We now use them instead of the literals that were defined in WildFly though
* enhancement of REST Client integration, this ones boils down to making HealthCheckResponse a concrete class instead of an abstract class. 

Both of these cases do not require additional tests in WildFly test suite.

Reply:
> annotation literals is mostly an implementation detail that does not accept consumer of the API. We now use them instead of the literals that were defined in WildFly though

Ok, so this could actually be mentioned in the "Implementation Plan" section, couldn't it?

Reply:
done

Reply:
done